src/columnar_transposition.rs

```diff
@@ -131,17 +191,21 @@ impl Cipher for ColumnarTransposition {
         }
 
         let mut plaintext = String::new();
-        // Okay this can be messy as the columns may be of unequal length
-        // Iterate over the headers of the columns
         for i in 0..max_col_size {
             for chr in self.key.chars() {
+                // Outer getting the key char
                 if let Some(column) = key.iter().find(|x| x.0 == chr) {
-                    // TODO: Fix is the columns are uneven
-                    //  Currently breaks the decryption
-                    // Also, this breaks when whitespace is added to end of string
-                    // if i < column.1.len() {
-                    plaintext.push(column.1[i]);
-                // }
+                    if i < column.1.len() {
+                        let c = column.1[i];
+                        // Special case for whitespace as the nulls can be trimmed
+                        if self.use_nulls {
```

Comment:
For example, you could avoid using the variable `use_nulls`, using the `if let` pattern in Rust....
This could look something like:
```
if let Some(n_char) = self.null_char {
   if c == n_char && !c.is_whitespace() {
      break;
   }
}
```